jest/private/bazel_snapshot_resolver.cjs

```diff
@@ -45,53 +50,42 @@ let snapshotResolver = process.env.JEST_TEST__USER_SNAPSHOT_RESOLVER
   ? require(process.env.JEST_TEST__USER_SNAPSHOT_RESOLVER)
   : defaultSnapshotResolver;
 
-if (process.env.JEST_TEST__UPDATE_SNAPSHOTS_MODE) {
-  // This run of Jest is meant for generating reference snapshots for the snapshots update Bazel target
-  if (process.env.JEST_TEST__UPDATE_SNAPSHOTS_MODE == "directory") {
-    // If we're updating snapshots with Bazel then switch the directory that snapshots are written
-    // to to REFERENCE_SNAPSHOT_DIRECTORY so the generated snapshot path is different from the
-    // source snapshot path allowing us to use write_source_files to for the update target.
-    const origSnapshotResolver = snapshotResolver;
-    snapshotResolver = {
-      ...origSnapshotResolver,
-
-      resolveSnapshotPath: (testPath) => {
-        const orig = origSnapshotResolver.resolveSnapshotPath(testPath);
-        return path.join(
-          path.dirname(orig),
-          REFERENCE_SNAPSHOT_DIRECTORY,
-          path.basename(orig)
-        );
-      },
-
-      resolveTestPath: (snapshotPath) => {
-        const orig = origSnapshotResolver.resolveTestPath(snapshotPath);
-        return path.join(path.dirname(path.dirname(orig)), path.basename(orig));
-      },
-    };
-  } else if (process.env.JEST_TEST__UPDATE_SNAPSHOTS_MODE == "files") {
-    // If we're updating snapshots with Bazel then append a REFERENCE_SNAPSHOT_SUFFIX to the
-    // snapshot path so the generated snapshot path is different from the source snapshot path
-    // allowing us to use write_source_files to for the update target.
-    const origSnapshotResolver = snapshotResolver;
-    snapshotResolver = {
-      ...origSnapshotResolver,
-
-      resolveSnapshotPath: (testPath) =>
-        origSnapshotResolver.resolveSnapshotPath(testPath) +
-        REFERENCE_SNAPSHOT_SUFFIX,
-
-      resolveTestPath: (snapshotPath) =>
-        origSnapshotResolver.resolveTestPath(
-          snapshotPath.substr(
-            0,
-            snapshotPath.length - REFERENCE_SNAPSHOT_SUFFIX.length
-          )
-        ),
-    };
-  } else {
-    throw new Error(INTERNAL_ERROR_MSG);
-  }
-}
+// This run of Jest is meant for generating reference snapshots for the snapshots update Bazel target.
+
+// Switch the directory of snapshots to within the src dir (BUILD_WORKSPACE_DIRECTORY) instead
+// of the snapshots dir in bazel-bin.
+//
+// This means none of the snapshots copied into the bin dir will be used, all will be considered
+// snapshot test failures and be written to BUILD_WORKSPACE_DIRECTORY.
+const origSnapshotResolver = snapshotResolver;
+snapshotResolver = {
+  ...origSnapshotResolver,
+
+  resolveSnapshotPath: (testPath) => {
+    let orig = origSnapshotResolver.resolveSnapshotPath(testPath);
+
+    if (path.isAbsolute(orig)) {
+      orig = path.join(
+        process.env.BUILD_WORKSPACE_DIRECTORY,
+        path.relative(process.cwd(), orig)
```

Comment:
hmmm. this is probably going to be a problem if there is a `chdir`. I think you need a test case to handle that.

Reply:
Fixed, see https://github.com/aspect-build/rules_jest/compare/4dd40f91d5213de63146d500b2ae73700c7833ac..a9d43790eba6ba1a768281332ab49879408eddaa